src/bucket/BucketIndexImpl.cpp

```diff
@@ -450,6 +443,39 @@ BucketIndexImpl<IndexT>::getPoolshareTrustlineRange(
     return std::make_pair(startOff, endOff);
 }
 
+template <class IndexT>
+std::optional<std::pair<std::streamoff, std::streamoff>>
+BucketIndexImpl<IndexT>::getPoolshareTrustlineRange(
+    AccountID const& accountID) const
+{
+    // Get the smallest and largest possible trustline keys for the given
+    // accountID
+    auto upperBound = getDummyPoolShareTrustlineKey(
+        accountID, std::numeric_limits<uint8_t>::max());
+    auto lowerBound = getDummyPoolShareTrustlineKey(
+        accountID, std::numeric_limits<uint8_t>::min());
+
+    return getOffsetBounds(lowerBound, upperBound);
+}
+
+template <class IndexT>
+std::optional<std::pair<std::streamoff, std::streamoff>>
+BucketIndexImpl<IndexT>::getOfferRange() const
+{
+    // Get the smallest and largest possible offer keys
+    LedgerKey upperBound(OFFER);
+    upperBound.offer().sellerID.ed25519().fill(
+        std::numeric_limits<uint8_t>::max());
+    upperBound.offer().offerID = std::numeric_limits<int64_t>::max();
+
+    LedgerKey lowerBound(OFFER);
+    lowerBound.offer().sellerID.ed25519().fill(
+        std::numeric_limits<uint8_t>::min());
+    lowerBound.offer().offerID = std::numeric_limits<int64_t>::min();
```

Comment:
The minimum possible `offerID` is 0, not `INT64_MIN`, but this doesn't matter in practice because `getOffsetBounds` does a binary search using this `lowerBound`. Is my understanding correct?

Reply:
Correct, we just want the smallest possible dummy OFFER key so that we can properly seek the the first place in the file where the OFFER region starts. Whether or not the dummy key we use is actually valid doesn't matter.